backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java

```diff
@@ -371,6 +376,9 @@ public void tx(Transaction transaction) {
             });
         // only when commit succeeds we copy state to 'this'
         this.sync(db);
+        if (!Objects.equals(db.getActiveUser(), getActiveUser())) {
+          this.getListener().userChanged();
```

Comment:
this is actually different because in context of a tx the userChanged will not result in a call to the listener, therefore that needs to be done at this point. There are two reasons to refresh caches. (1) user changed (impact only on current session) and (2) schema/roles changed (impact on all sessions)